napkin_graph.py

```diff
@@ -314,9 +314,10 @@ def remove_duplicate_edges(self) -> None:
 	def create_id_to_raw(self) -> 'dict[int, str]':
 		""" Creates a dictionary mapping node ids to their raw code
 		"""
+		id_to_json, json_to_id = self.create_index_to_json_dict()
 		id_to_raw = {}
 		for node in self.nodes:
-			id_to_raw[node.id] = node.raw
+			id_to_raw[json_to_id[node.id]] = node.raw
```

Comment:
I think this is an artifact of our original code but if we're creating a new dictionary by indexing another dictionary at a value and re-assigning it, it's just too complicated and we should redesign this. The function `create_index_to_json_dict` should ideally be deleted and we need to find another way of implementing or fixing our indexing problem.